pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -55,7 +57,8 @@
 
 # The type of build to perform. This is passed to `dotnet` with the `--configuration` flag. Possible values are `Release`, `Debug`, etc.
 , buildType ? "Release"
-# If set to true, builds the application as a self-contained - removing the runtime dependency on dotnet
+# If set to true, builds the application as a self-contained - removing the runtime dependency on dotnet.
+# Use null if the dotnet version you need does not support the `--self-contained` (or `--no-self-contained`) flag.
```

Comment:
Please try with https://github.com/NixOS/nixpkgs/pull/185021, the changes in that PR should fix this flag for dotnet <6. Then, this `null` logic can be simplified.